step3/main.swift

```diff
@@ -7,47 +7,70 @@
 
 import Foundation
 
-let iceLength = Int(readLine()!)!
-let iceShape = readLine()!
-let toppingChoice = readLine() ?? " "
-let barSize = Int(readLine()!)!
+print("길이를 입력하세요 !")
+let iceLengthTypeStr = readLine() ?? "0"
+print("몸통을 입력하세요 !")
+var iceShape: String = readLine() ?? ""
+print("토핑을 입력하세요 !")
+var toppingChoice: String = readLine() ?? ""
+print("막대길이를 입력하세요 !")
+let barSizeTypeStr = readLine() ?? "0"
+
+let iceLength: Int = Int(iceLengthTypeStr) ?? 0
+let barSize: Int = Int(barSizeTypeStr) ?? 0
+
 var toppingCount = 0 // 토핑의 갯수를 세줌. 비어있어도 2
-checkTopping()
+var iceCount = 0
 
-let barShape = ["| |", "||", "|"]
-let checkIceSize = iceShape.count + toppingCount
+countTopping()
+countIceSize()
 
+let barShape = ["| |", "||", "|"]
+let checkIceSize = iceCount + toppingCount
 
-print("<정보>")
-print("길이 : \(iceLength)\n몸통 : \(iceShape)\n토핑 : \(toppingChoice)\n막대길이 : \(barSize)")
-createShape(x: iceLength)
-createBar(y: barSize)// (checkIceSize - 3) / 2 = 앞에 필요한 여백
+let printInfo = """
+<정보>
+길이 : \(iceLength)\n몸통 : \(iceShape)\n토핑 : \(toppingChoice)\n막대길이 : \(barSize)
+"""
+print(printInfo)
+createShape(countLength: iceLength)
+createBar(countLength: barSize)// (checkIceSize - 3) / 2 = 앞에 필요한 여백
 
 
-func createShape(x: Int){
+func createShape(countLength: Int){
     if toppingChoice == "" {
-        for _ in 0...x-1{
+        for _ in 0...countLength-1{
             print(" \(iceShape) ")
         }
     } else {
-        for _ in 0...x-1{ // iceLength
-        print("\(toppingChoice)\(iceShape)\(toppingChoice)")
+        if iceShape == "|0|"{
+            for i in 0...countLength-1{
+                if i % 2 == 0{
```

Comment:
이렇게 해도 괜찮을 거 같은데요?
```suggestion
                if count % 2 == 0 {
```
또는
```suggestion
                if index % 2 == 0 {
```